Content.Server/Xenoarchaeology/XenoArtifacts/ArtifactSystem.cs

```diff
@@ -296,4 +315,27 @@ public ArtifactNode GetRootNode(List<ArtifactNode> allNodes)
     {
         return allNodes.First(n => n.Depth == 0);
     }
+
+    // SS220-BonusForFullyDiscovered - start
+    
+    private void SpawnBonus(EntityUid uid, ArtifactComponent component)
+    {
+        if (component.BonusProtype == null)
+            return;
+
+        TrySpawnNextTo(component.BonusProtype[_random.Next(component.BonusProtype.Count - 1)], uid, out var protoEnt);
+
+        if (protoEnt == null)
+            return;
+
+        var protoUid = (EntityUid)protoEnt;
+
+        var xform = Transform(protoUid);
+        var throwing = xform.LocalRotation.ToWorldVec() * 5f; // magic number throwing force
+        var direction = xform.Coordinates.Offset(throwing);
+
+        _throwing.TryThrow(protoUid, direction);
```

Comment:
```suggestion
        var protoId = _random.Pick(component.BonusPrototype);
        if (!TrySpawnNextTo(protoId, uid, out var protoUid);
            return;

        var xform = Transform(protoUid.Value);
        var throwing = xform.LocalRotation.ToWorldVec() * 5f; // magic number throwing force
        var direction = xform.Coordinates.Offset(throwing);

        _throwing.TryThrow(protoUid.Value, direction);
```
Вроде правильно написал (через гит не совсем удобно).
Проверка ProtoUid на null - лишняя, т.к. `TrySpawnNextTo` принимает значение true только когда выдаваемый в out EntityUid не null